src/Doctrine/Extensions/DBAL/Types/UTCDateTimeImmutableType.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace App\Doctrine\Extensions\DBAL\Types;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\ConversionException;
+use Doctrine\DBAL\Types\DateTimeImmutableType;
+
+class UTCDateTimeImmutableType extends DateTimeImmutableType
+{
+    private static ?\DateTimeZone $utc;
+
+    public function convertToDatabaseValue($value, AbstractPlatform $platform): ?string
+    {
+        if ($value instanceof \DateTimeImmutable) {
+            if (self::getUtc()->getName() !== $value->getTimezone()->getName()) {
+                $mutable = \DateTime::createFromImmutable($value);
+                $mutable->setTimezone(self::getUtc());
+
+                $value = \DateTimeImmutable::createFromMutable($mutable);
```

Comment:
Why don't you just use

```suggestion
                $value = $value->setTimezone(self::getUtc());
```

?